### PR TITLE
[master] Use Substrate 2.0

### LINKED ISSUE #27: Can we support substrate v2.0?
I've been working to tweak the repo and make it compatible against substrate master branch, I don't know if you have any plan to support the new api

### PR DESCRIPTION
**Pull request by danforbes**
Along with #32, this closes #27.

### CODE REVIEW

File: src/chain_spec.rs [hunk 1]
```diff
@@ -1,142 +1,143 @@
...
-                None,
-                None,
-                None,
-                None,
-            ),
-        })
-    }
+	/// Get an actual chain config from one of the alternatives.
+	pub(crate) fn load(self) -> Result<ChainSpec, String> {
+		Ok(match self {
```
Comment (nczhu) [inline src/chain_spec.rs]: Nit: Indents look off here

File: src/chain_spec.rs [hunk 2]
```diff
@@ -1,142 +1,143 @@
...
-) -> GenesisConfig {
-    GenesisConfig {
-		consensus: Some(ConsensusConfig {
-			code: include_bytes!("../runtime/wasm/target/wasm32-unknown-unknown/release/utxo_runtime_wasm.compact.wasm").to_vec(),
-			authorities: initial_authorities.clone(),
-		}),
-		system: None,
-		timestamp: Some(TimestampConfig {
-			minimum_period: 5, // 10 second block time.
+fn testnet_genesis(initial_authorities: Vec<(AuraId, GrandpaId)>,
```
Comment (nczhu) [inline src/chain_spec.rs]: Thanks for the refactoring here

File: src/chain_spec.rs (hunk 1 above)
Comment (danforbes) [reply src/chain_spec.rs]: Okay, I _believe_ that I have all the indentation issues taken care of (at least in this file) and definitely appreciate the close eye. I was having some issues with the genesis configuration (I wasn't able to see an account being seeded with UTXOs) and I think now that the diff is a little more clear I am able to see why.

File: src/chain_spec.rs [hunk 3]
```diff
@@ -94,49 +114,30 @@ impl Alternative {
...
+      sudo: Some(SudoConfig {
+        key: root_key,
+      }),
+      aura: Some(AuraConfig {
+        authorities: initial_authorities.iter().map(|x| (x.0.clone())).collect(),
+      }),
+      grandpa: Some(GrandpaConfig {
+        authorities: initial_authorities.iter().map(|x| (x.1.clone(), 1)).collect(),
+      }),
+    }
```
Comment (nczhu) [inline src/chain_spec.rs]: Issue: your new chain_spec breaks the testnet/ui demo instructions in README section #demo-polkadot-UI. 

I agree with this refactoring (thank you) - it makes the repo itself much cleaner.

However, Alice/other accounts are no longer preloaded with any utxo at genesis, which the UI demo assumes.

Suggestion: also update the demo readme to make this tutorial work with the new workshop set up. i.e. just make the user sudo -> mint into Alice, etc. Note that then, all other steps and the encoded `hints` must also be updated.

**Comment by nczhu** [comment]
The last test is no longer passing, can you double check everything: 

```
running 8 tests
test utxo::tests::attack_by_double_counting_input ... ok
test utxo::tests::attack_by_overflowing ... ok
test utxo::tests::attack_by_over_spending ... ok
test utxo::tests::attack_by_double_generating_output ... ok
test utxo::tests::attack_by_permanently_sinking_outputs ... ok
test utxo::tests::attack_with_empty_transactions ... ok
test utxo::tests::attack_with_invalid_signature ... ok
test utxo::tests::valid_transaction ... FAILED
```

**Comment by danforbes** [comment]
Good looking out...got that test passing. Please note that I am still putting the finishing touches on this PR...just getting rid of weird whitespace diffs and going through the docs. I will try to have it complete today. Thanks again for all your help!

**Pull request merged**